### ISSUE TITLE
Founder appears as separate user

### ISSUE THREAD

**Issue by iptq** [oragono/oragono#298]
Seems like I'm showing up twice, once as `michael` and once as `~michael` when that's actually just me as the founder. The channel reports that the nicks are different when I checked. I triggered this by registering the channel and then rejoining the channel. Did I do something wrong or is this behavior unexpected?

![image](https://user-images.githubusercontent.com/1700346/47834085-f5360f00-dd6b-11e8-82c9-13bcf499f903.png)

**Comment by DanielOaks** [comment]
Hmm, this sounds quite strange. Just curious, what client are you using? I'll take a look and see if I can reproduce this issue, thanks for the report!

**Comment by iptq** [comment]
I'm using weechat, let me know if you need my configs or anything else.

**Comment by slingamn** [comment]
If you have time, the raw protocol input and output might help: https://weechat.org/blog/post/2015/08/24/Hex-dump-in-IRC-raw-buffer

**Comment by iptq** [comment]
Looks like this is a weechat problem, I'll close this for now

**Comment by cyruslopez** [comment]
I don't think this is a weechat-only problem as Textual suffers from the same issue.

**Comment by slingamn** [comment]
@iptq here's where we're at with this: I can't find any incorrect responses sent by oragono at the wire level, but the behavior (duplicated nicknames in the channel membership list, one with a `~` prefix) has been seen with weechat, Textual, and irssi.

Were you able to clarify that this is not an oragono issue? Did this get filed as a weechat bug?

**Comment by DanielOaks** [comment]
@slingamn just curious, have we either captured the raw traffic from the server-side or sat [something](https://github.com/goshuirc/ircdog#ircdog) in the middle of Ora and weechat/textual/irssi/etc and seen exactly what those clients do upon joining the channel? from my looks there's nothing wrong with how we're sending out the on-join responses, but maybe those clients do e.g. a `NAMES` or some other command that we're returning incorrect responses to. that many broken clients signals that it's likely something weird on our end, maybe mixed with a bit of clients being less than 1000% safe with how they process incoming numerics.

**Comment by slingamn** [comment]
Yeah, we've looked at the raw traffic and it looks correct. This is redacted:

````
:server.name 353 joiner = #chname :~joiner ~OtherFounder
:server.name 366 joiner #chname :End of NAMES list
````

The one thing I can think of is that these clients are expecting their own nick to be unprefixed in the `353 RPL_NAMREPLY` line, and only recognize their owner/founder status when they receive a subsequent `MODE +q` line. So when they get their name with the `~` prefix in the 353 line, they get confused?

**Comment by slingamn** [comment]
I tried to reproduce this with irssi trunk (1c6f2560989f4), weechat trunk (d6e4bbf24447), and the latest binary release of Textual (7.1.0). I couldn't reproduce it in any of these clients via the following procedure: connect, authenticate to an account with NickServ, then join a channel of which that account is the registered founder.

If this can still be reproduced, it would help to have a complete dump of the i/o from the oragono side, e.g., with this logging config:

````yaml
logging:
    -
        method: stderr
        type: "*"
        level: debug
````

**Comment by slingamn** [comment]
I'm going to close this because we couldn't reproduce it and in any case it may have been fixed by #353
